kernel_gateway/gatewayapp.py

```diff
@@ -312,19 +320,23 @@ def init_configurables(self):
                     self.prespawn_count, self.max_kernels)
                 )
 
-        if self.api == 'notebook-http':
-            self.kernel_pool = ManagedKernelPool(
-                self.prespawn_count,
-                self.kernel_manager
-            )
-        else:
-            self.kernel_pool = KernelPool(
-                self.prespawn_count,
-                self.kernel_manager
-            )
+        
+        api_module = self._load_api_module(self.api)
+        func = getattr(api_module, 'create_personality', None)
+        if func is None:
+            self.log.critical('ERROR: the personality factory function was not found in the API module {}'.format(self.api))
+            self.exit(1)
+        self.personality = func(self, self.kernel_manager.seed_kernelspec)
+
+        self.personality.init_configurables(self.log, {
+            'allow_notebook_download': self.allow_notebook_download,
+            'prespawn_count': self.prespawn_count,
+            'kernel_manager': self.kernel_manager,
+            'base_url': self.base_url,
+            'seed_uri': self.seed_uri})
```

Comment:
These might build up over time. How about just passing `self` and letting the personality access whatever it needs from the application instance?